openssl/src/x509/mod.rs

```diff
@@ -64,6 +64,41 @@ foreign_type! {
     pub struct X509StoreContextRef;
 }
 
+impl X509StoreContext {
+    pub fn new() -> Result<X509StoreContext, ErrorStack> {
+        unsafe {
+            ffi::init();
+            cvt_p(ffi::X509_STORE_CTX_new()).map(|p| X509StoreContext(p))
+        }
+    }
+
+    /// Initializes the store context to verify the certificate.
+    ///
+    /// This Context can only be used once, subsequent to any validation, the context must be reinitialized.
+    ///
+    /// # Arguments
+    ///
+    /// * `trust` - a store of the trusted chain of certificates, or CAs, to validated the certificate
+    /// * `cert` - certificate to validate
+    /// * `cert_chain` - the certificates chain
+    pub fn init(&self, trust: &store::X509StoreRef, cert: &X509Ref, cert_chain: &StackRef<X509>) -> Result<(), ErrorStack> {
+        unsafe {
+            cvt(ffi::X509_STORE_CTX_init(self.as_ptr(), trust.as_ptr(), cert.as_ptr(), cert_chain.as_ptr()))
+                .map(|_| ())
+        }
+    }
+
+    /// Verifies the certificate associated in the `init()` method
+    ///
+    /// This consumes self as the `X509StoreContext` must be reinitialized subsequent to any cally to verify. 
+    pub fn verify_cert(&self) -> Result<Option<X509VerifyError>, ErrorStack> {
```

Comment:
thanks. apparently the tests failed, but for some reason all of mine passed locally... huh.